Gitweb:     http://git.kernel.org/git/?p=linux/kernel/git/torvalds/linux-2.6.git;a=commit;h=12c22d6ef299ccf0955e5756eb57d90d7577ac68
Commit:     12c22d6ef299ccf0955e5756eb57d90d7577ac68
Parent:     05dda977f2574c3341abef9b74c27d2b362e1e3a
Author:     Linus Torvalds <torvalds@linux-foundation.org>
AuthorDate: Wed Mar 26 11:22:40 2008 -0700
Committer:  Linus Torvalds <torvalds@linux-foundation.org>
CommitDate: Wed Mar 26 11:22:40 2008 -0700

    Revert "PCI: remove transparent bridge sizing"
    
    This reverts commit 8fa5913d54f3b1e09948e6a0db34da887e05ff1f, which
    caused various interesting problems for people, including wrong resource
    allocations.  See for example bugzilla entry "2.6.25-rc2: ohci1394
    problem (MMIO broken)" at
    
    	http://bugzilla.kernel.org/show_bug.cgi?id=10080
    
    And Gary Hade says:
    
     "The same change had also exposed an issue reported by Paul Martin that
      has been causing an Oops while hotplugging ThinkPads to a ThinkPad
      Dock II.  See
    
    	http://lkml.org/lkml/2008/2/19/405
    	http://bugzilla.kernel.org/show_bug.cgi?id=9961
    
      I have a fix for the ThinkPad docking Oops but if the issue being
      discussed here is caused by the transparent bridge sizing removal
      change I totally agree that it should be reverted."
    
      The transparent bridge sizing removal change was motivated by
      insufficient PCI memory resource for a transparent bridge window that
      was being created as a result of expansion ROM(s) being included in
      the transparent bridge sizing calculations.
    
      A later "PCI: Remove default PCI expansion ROM memory allocation"
      change ( re: http://lkml.org/lkml/2007/12/11/361 ) removes the
      expansion ROM(s) from the transparent bridge sizing calculations which
      actually resolves the original issue in a different manner.  So, even
      if the "PCI: remove transparent bridge sizing" is not problematic it
      is no longer needed anyway."
    
    Identified-by: Ivan Kokshaysky <ink@jurassic.park.msu.ru>
    Tested-by: Thomas Meyer <thomas@m3y3r.de>
    Acked-by: Gary Hade <garyhade@us.ibm.com>
    Acked-by: Ingo Molnar <mingo@elte.hu>
    Cc: Stefan Richter <stefanr@s5r6.in-berlin.de>
    Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>
---
 drivers/pci/setup-bus.c |    5 -----
 1 files changed, 0 insertions(+), 5 deletions(-)

diff --git a/drivers/pci/setup-bus.c b/drivers/pci/setup-bus.c
index 125e7b7..f7cb8e0 100644
--- a/drivers/pci/setup-bus.c
+++ b/drivers/pci/setup-bus.c
@@ -486,12 +486,7 @@ void __ref pci_bus_size_bridges(struct pci_bus *bus)
 		break;
 
 	case PCI_CLASS_BRIDGE_PCI:
-		/* don't size subtractive decoding (transparent)
-		 * PCI-to-PCI bridges */
-		if (bus->self->transparent)
-			break;
 		pci_bridge_check_ranges(bus);
-		/* fall through */
 	default:
 		pbus_size_io(bus);
 		/* If the bridge supports prefetchable range, size it
--
To unsubscribe from this list: send the line "unsubscribe git-commits-head" in
the body of a message to majordomo@vger.kernel.org
More majordomo info at  http://vger.kernel.org/majordomo-info.html


